src/tools/melt/melt.cs

```diff
@@ -308,6 +308,30 @@ private static void MeltBinaryTable(Pdb inputPdb, InstallPackage package, string
             }
         }
 
+        private static bool ValidateMSIMatchesPdb(InstallPackage package, Pdb inputPdb)
+        {
+            const string KEY_PRODUCT_CODE = "ProductCode";
```

Comment:
By declaring `const` you're actually causing the string to be allocated multiple times. Either declare as `static readonly` at the class scope or declare here without `const` so the same string instance is used.